include/kllvm/codegen/CreateTerm.h

```diff
@@ -19,71 +19,95 @@ class CreateTerm {
+llvm::StructType *getBlockType(
+    llvm::Module *Module, KOREDefinition *definition, const KORESymbol *symbol);
+llvm::Value *getBlockHeader(
+    llvm::Module *Module, KOREDefinition *definition, const KORESymbol *symbol,
+    llvm::Type *BlockType);
 
-/* returns the llvm::Type corresponding to the type of the result of calling createTerm
-   on the specified pattern. */
-ValueType termType(KOREPattern *pattern, llvm::StringMap<ValueType> &substitution, KOREDefinition *definition);
+/* returns the llvm::Type corresponding to the type of the result of calling
+   createTerm on the specified pattern. */
+ValueType termType(
+    KOREPattern *pattern, llvm::StringMap<ValueType> &substitution,
+    KOREDefinition *definition);
 
-/** creates a function that applies the specified rule once it has matched, and returns the name of the function. */
-void makeApplyRuleFunction(KOREAxiomDeclaration *axiom, KOREDefinition *definition, llvm::Module *Module, bool bigStep = false);
-std::string makeApplyRuleFunction(KOREAxiomDeclaration *axiom, KOREDefinition *definition, llvm::Module *Module, std::vector<Residual> residuals);
-/** creates a function that evaluates the side condition of the specified rule, and returns the name of the function. Returns empty string if function has no side condition. */
-std::string makeSideConditionFunction(KOREAxiomDeclaration *axiom, KOREDefinition *definition, llvm::Module *Module);
+/** creates a function that applies the specified rule once it has matched, and
+ * returns the name of the function. */
+void makeApplyRuleFunction(
+    KOREAxiomDeclaration *axiom, KOREDefinition *definition,
+    llvm::Module *Module, bool bigStep = false);
+std::string makeApplyRuleFunction(
+    KOREAxiomDeclaration *axiom, KOREDefinition *definition,
+    llvm::Module *Module, std::vector<Residual> residuals);
+/** creates a function that evaluates the side condition of the specified rule,
+ * and returns the name of the function. Returns empty string if function has no
+ * side condition. */
+std::string makeSideConditionFunction(
+    KOREAxiomDeclaration *axiom, KOREDefinition *definition,
+    llvm::Module *Module);
 
 /* returns the llvm::Type corresponding to the specified KORE sort category */
 llvm::Type *getValueType(ValueType sort, llvm::Module *Module);
 llvm::Type *getParamType(ValueType sort, llvm::Module *Module);
 
 void addAbort(llvm::BasicBlock *block, llvm::Module *Module);
 
-llvm::Value *allocateTerm(llvm::Type *AllocType, llvm::BasicBlock *block, const char *allocFn = "koreAlloc");
-llvm::Value *allocateTerm(llvm::Type *AllocType, llvm::Value *Len, llvm::BasicBlock *block, const char *allocFn = "koreAlloc");
-}
+llvm::Value *allocateTerm(
+    llvm::Type *AllocType, llvm::BasicBlock *block,
+    const char *allocFn = "koreAlloc");
+llvm::Value *allocateTerm(
+    llvm::Type *AllocType, llvm::Value *Len, llvm::BasicBlock *block,
+    const char *allocFn = "koreAlloc");
+} // namespace kllvm
```

Comment:
Declarations and definitions will get wrapped to respect the maximum line length. It's possible to specify one line per parameter if that's preferable